### PR TITLE
User can empty inbox

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT](https://www.pivotaltracker.com/story/show/170096461)

Checks functionality of moving message to trash

### CODE REVIEW

File: features/step_definitions/background_steps.rb [hunk 1]
```diff
@@ -10,4 +10,11 @@
   table.hashes.each do |hash|
     create(:user, hash)
   end
-end
\ No newline at end of file
+end
+
+  Given("the inbox has content") do
```
Comment (oliverochman) [inline features/step_definitions/background_steps.rb]: fix indentation

File: features/user_can_delete_message.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  I want to be able to delete messages
+
+  Background:
+    Given the following user exist:
+
+      | name   | email          | password  | id |
+      | dmybob | dmybob@bob.bob | bobbobbob | 1  |
+      | dmypip | dmypip@bob.bob | pippippip | 2  |
+
+    Given the inbox has content
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: please change to `And the inbox has content`

File: features/user_can_delete_message.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  Scenario: Moves message to trash
+    Given I am on the Login page
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: I would put this in the background step

File: features/user_can_delete_message.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    And I fill in "Email" with "dmypip@bob.bob"
+    And I fill in "Password" with "pippippip"
+    And I click on "Log in"
+    And I click on "Inbox"
+    And I click on "View"
+    Then I click on "Move to trash"
+    And I click on "Trash"
+    Then I should see "For ever yours"
+
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: Can a user remove a message that he has sent?

File: features/user_can_delete_message.feature (hunk 2 above)
Comment (mantanquzzmo) [reply features/user_can_delete_message.feature]: ok

File: features/user_can_delete_message.feature (hunk 3 above)
Comment (mantanquzzmo) [reply features/user_can_delete_message.feature]: so would we

File: features/user_can_delete_message.feature (hunk 4 above)
Comment (mantanquzzmo) [reply features/user_can_delete_message.feature]: yes it can. tested.

File: features/user_can_delete_message.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
+Feature: User can empty inbox
+  As a user
+  In order to clean up my inbox
+  I want to be able to delete messages
+
+  Background:
+    Given the following user exist:
+
+      | name   | email          | password  | id |
+      | dmybob | dmybob@bob.bob | bobbobbob | 1  |
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: please make the users more descriptive
e.g
user_1@mail.com, password etc.

File: features/user_can_delete_message.feature [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      | dmypip | dmypip@bob.bob | pippippip | 2  |
+
+    Given the inbox has content
+    And I am on the Login page
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: I would like you guys to login with the warden helper that is being used in the `user_can_compose_message` feature test

File: features/user_can_delete_message.feature [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  Scenario: Moves received message to trash
+    Then I fill in "Email" with "dmypip@bob.bob"
+    And I fill in "Password" with "pippippip"
+    And I click on "Log in"
+    And I click on "Inbox"
+    And I click on "View"
+    Then I click on "Move to trash"
+    And I click on "Trash"
+    Then I should see "For ever yours"
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: there is not supposed to to be any `Then` before this assertion

File: features/user_can_delete_message.feature (hunk 6 above)
Comment (mantanquzzmo) [reply features/user_can_delete_message.feature]: ok

File: features/user_can_delete_message.feature (hunk 5 above)
Comment (mantanquzzmo) [reply features/user_can_delete_message.feature]: ok

File: features/user_can_delete_message.feature [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  I want to be able to delete messages
+
+  Background:
+    Given the following user exist:
+
+      | name   | email          | password | id |
+      | dummy1 | user1@mail.com | password | 1  |
+      | dummy2 | user2@mail.com | password | 2  |
+
+    Given the inbox has content
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: please change to `And the inbox has content`

File: features/user_can_delete_message.feature [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    And I click on "Move to trash"
+    And I click on "Trash"
+    Then I should see "For ever yours"
+
+  Scenario: Moves sent message to trash
+    And I go to the sent page
+    And I click on "View"
+    And I click on "Move to trash"
+    And I click on "Trash"
+    Then I should see "For ever yours"
```
Comment (oliverochman) [inline features/user_can_delete_message.feature]: Can you make the assertion a bit different here, maybe change the subject of the message that the user sent from the one in their inbox?

**Pull request merged**